src/messaging/ExchangeContext.cpp

```diff
@@ -266,6 +267,30 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
     }
 }
 
+CHIP_ERROR ExchangeContext::SendStatusReport(const Protocols::SecureChannel::StatusReport & statusReport)
```

Comment:
Why need this? A `SendMessage` should be good enough. THere is no need for this additional method

Reply:
Because actually producing the packet buffer to send is a bit of a pain.  And this way that code only needs to exist in one place instead of being duplicated in multiple places.

Reply:
@tcarmelveilleux Please take a look at https://github.com/project-chip/connectedhomeip/issues/27473. Different protocols sends status report and construct it themselves, so the idea was to have a method which can be used by all to reduce the code duplication.

Reply:
I agree about making code all in one place, but right now the payload is not constructed in a single place. See my other comment: the code can be combined for what is in CASESession and what is in ExchangeContext on this PR, into code in the StatusRepory module.

Reply:
As per the discussion with Tennessee, Added the builder `StatusReport::MakeBusyStatusReportMessage(waitTime)` and unit tests for the same.

For moving the common code from builder and `PairingSession::SendStatusReport`, will do a followup. (https://github.com/project-chip/connectedhomeip/issues/28471)